tests/test_environement.py

```diff
@@ -0,0 +1,160 @@
+########
+# Copyright (c) 2017 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import os
+
+import aria
+import pytest
+
+from aria.storage.filesystem_rapi import FileSystemResourceAPI
+from aria.storage.sql_mapi import SQLAlchemyModelAPI
+from aria_plugin import environment, constants, utils, exceptions
+
+
+class TestEnvironment(object):
+
+    @pytest.fixture
+    def env(self, mocker):
+        self._workdir = 'workdir_path'
+        mocker.patch('aria_plugin.utils.silent_create',
+                     return_value=self._workdir)
+        mocked_ctx = mocker.MagicMock()
+        return environment.Environment(mocked_ctx)
+
+    def test_ctx_logger(self, env):
+        env.ctx_logger.info('some info')
+        env._ctx.logger.info.assert_called_once_with('some info')
+
+    def test_workdir(self, env):
+        assert env.workdir == self._workdir
+
+    def test_blueprint_dir(self, env, mocker):
+        env._ctx.tenant_name = 'tenant_name'
+        env._ctx.blueprint.id = 'blueprint_id'
+        mocker.patch('os.path.join')
+
+        blueprint_dir = env.blueprint_dir
+
+        os.path.join.assert_called_once_with(
+            mocker.ANY, 'tenant_name', 'blueprint_id')
+
+    def test_model_storage_dir(self, env):
+        assert env.model_storage_dir == os.path.join(self._workdir, 'models')
+
+    def test_model_resource_storage_dir(self, env):
+        assert env.resource_storage_dir == os.path.join(
+            self._workdir, 'resources')
+
+    def test_model_plugins_dir(self, env):
+        assert env.aria_plugins_dir == os.path.join(self._workdir, 'plugins')
+
+    def test_model_storage(self, env, mocker):
+        mocker.patch('aria.application_model_storage')
+
+        model_storage = env.model_storage
+
+        aria.application_model_storage.assert_called_once_with(
+            api=SQLAlchemyModelAPI,
+            initiator_kwargs={
+                'base_dir': os.path.join(self._workdir, 'models')
+            }
+        )
+
+        assert model_storage == env.model_storage
```

Comment:
explain that this line is to check that model_storage is the same object in every call